### ISSUE TITLE
[BUG] Unable to access harvester VIP nor node IP after reboot or fully power cycle node machines (Intermittent)

### ISSUE THREAD

**Issue by TachunLin** [harvester/harvester#1669]
**Describe the bug**

> In the following two types of scenarios, user maybe encounter the issue of `not able to access harvester dashboard by VIP` and `access to node machine IP` (ping, ssh) 

#### Scenario1: 
1. On a 3 nodes harvester cluster in `local kvm machine` when we reboot one of the node machine 

#### Scenario2: 
2. On a 3 nodes harvester cluster in `bare metal machines` when shutdown all node machines, wait 15 minutes above then power on each node by sequence. (20 seconds in between)

**Reproduce rate: Intermittent**

* Node 1 shows `Unknown` status
![image](https://user-images.githubusercontent.com/29251855/145829584-606b1d4b-613c-4e16-a7fb-7545c3ec6bc1.png)

* Can't access node 2, node3 and VIP (node 1 is available)
```
davidtclin@localhost:~/Documents/Software/qemu_kvm> ping 192.168.122.247
PING 192.168.122.247 (192.168.122.247) 56(84) bytes of data.
64 bytes from 192.168.122.247: icmp_seq=1 ttl=64 time=0.287 ms
64 bytes from 192.168.122.247: icmp_seq=2 ttl=64 time=0.323 ms
64 bytes from 192.168.122.247: icmp_seq=3 ttl=64 time=0.267 ms
64 bytes from 192.168.122.247: icmp_seq=4 ttl=64 time=0.302 ms
^C
--- 192.168.122.247 ping statistics ---
4 packets transmitted, 4 received, 0% packet loss, time 3045ms
rtt min/avg/max/mdev = 0.267/0.294/0.323/0.029 ms
davidtclin@localhost:~/Documents/Software/qemu_kvm> ping 192.168.122.130
PING 192.168.122.130 (192.168.122.130) 56(84) bytes of data.
From 192.168.122.1 icmp_seq=1 Destination Host Unreachable
From 192.168.122.1 icmp_seq=2 Destination Host Unreachable
From 192.168.122.1 icmp_seq=5 Destination Host Unreachable
From 192.168.122.1 icmp_seq=6 Destination Host Unreachable
From 192.168.122.1 icmp_seq=7 Destination Host Unreachable
^C
--- 192.168.122.130 ping statistics ---
7 packets transmitted, 0 received, +5 errors, 100% packet loss, time 6121ms
pipe 4
davidtclin@localhost:~/Documents/Software/qemu_kvm> ping 192.168.122.85
PING 192.168.122.85 (192.168.122.85) 56(84) bytes of data.
From 192.168.122.1 icmp_seq=1 Destination Host Unreachable
From 192.168.122.1 icmp_seq=2 Destination Host Unreachable
From 192.168.122.1 icmp_seq=3 Destination Host Unreachable
From 192.168.122.1 icmp_seq=4 Destination Host Unreachable
^C
--- 192.168.122.85 ping statistics ---
7 packets transmitted, 0 received, +4 errors, 100% packet loss, time 6131ms
pipe 4
davidtclin@localhost:~/Documents/Software/qemu_kvm> ping 192.168.122.142
PING 192.168.122.142 (192.168.122.142) 56(84) bytes of data.
From 192.168.122.1 icmp_seq=1 Destination Host Unreachable
From 192.168.122.1 icmp_seq=2 Destination Host Unreachable
From 192.168.122.1 icmp_seq=3 Destination Host Unreachable
^C
--- 192.168.122.142 ping statistics ---
5 packets transmitted, 0 received, +3 errors, 100% packet loss, time 4045ms
pipe 3
```

**To Reproduce**

#### Scenario1: 
1. Prepare a 3 nodes harvester cluster on local kvm machines
2. Enable virtual network on `harvester-mgmt` NIC
3. Create virtual network `vlan1` with id `1`
4. SSH to node 1, run `reboot`  as root
5. Wait for node 1 launch harvester
6. Try to access harvester dashboard by VIP 
7. Check node 1 accessibility
8. Reboot node2, node3 and node1
9. Try to access node 2, node 3 IP 
 

#### Scenario2: 

1. Prepare a 3 nodes harvester cluster on bare metal server
2. Enable virtual network on `harvester-mgmt` NIC
3. Create virtual network `vlan1` with id `1`
4. Provision a RKE2 cluster from remote ranhcer
5. Shutdown node 3 harvester machine, wait 10 seconds 
6. Shutdown node 2 harvester machine, wait 10 seconds
7. Shutdown node 1 harvester machine
8. Wait for 15 minutes or more
9. Power on node 1 harvester machine, wait 25 seconds
10. Power on node 2 harvester machine, wait 25 seconds
11. ower on node 3 harvester machine
12. Check 3 nodes harvester can back to work

**Expected behavior**

Since we can't access harvester nor run kubectl to get support bundle.
We try to get rke2 server log, hope can provide as much information as possible for troubleshooting

```
harvester-node01-100rc1:/home/rancher # kubectl get nodes
The connection to the server 127.0.0.1:6443 was refused - did you specify the right host or port?
```

* RKE2 server log for `node 1`  on provo bare machine 
 
  - [rke2-server_provo_node1.log](https://github.com/harvester/harvester/files/7698825/rke2-server_provo_node1.log)

* RKE2 server log for `node 1`  on local kvm machine
 
 - [rke2-server-local-node1.log](https://github.com/harvester/harvester/files/7704619/rke2-server-local-node1.log)

On 3 nodes harvester cluster, reboot any of the node machine or fully power cycle all machine should be able to access harvester dashboard by VIP and access to each node without connectivity issue.

**Support bundle**

**Scenario1 Test Environment:**
 - Harvester ISO version: master-5d6f3db0-head (12/11 after 1.0.0-rc1)
 - Underlying Infrastructure: 3 nodes harvester on Provo bare metal machines

**Scenario2 Test Environment:**
 - Harvester ISO version: v1.0.0-rc1
 - Underlying Infrastructure: 3 nodes harvester on local kvm machines

**Additional context**
Please refer to [issues/#1428/comments](https://github.com/harvester/harvester/issues/1428#issuecomment-991901048) for detailed issue description on bare metal machines .

**Area of effect**
This issue may prevent to reproduce and related to the following issues
1. #1561 
2. #1428

**Comment by yaocw2020** [comment]
For testing: (Use the master-head version of harvester network controller)
- Enable VLAN with NIC `harvester-mgmt`
- Create VLAN 1
- Disable VLAN and enable VLAN again
- Try to connect the hosts and they will be accessiable.

**Comment by noahgildersleeve** [comment]
Tested in master-3d096946-head in KVM virtualized environment. VIP stops working, but the node IP still works.

1. Enable VLAN with NIC harvester-mgmt
2. Create VLAN 1
3. Reboot system
4. Wait for system to be ready showing VIP
5. Try to go to web UI at VIP
[supportbundle_d8ce0447-ea2a-40b7-880d-937953c890b5_2021-12-15T18-42-59Z.zip](https://github.com/harvester/harvester/files/7721956/supportbundle_d8ce0447-ea2a-40b7-880d-937953c890b5_2021-12-15T18-42-59Z.zip)

**Comment by noahgildersleeve** [comment]
It's worth noting that if you do the same setup with VLAN 1 on a different NIC it works fine. It's just when you do it on the harvester-mgmt NIC.

**Comment by yaocw2020** [comment]
> Tested in master-3d096946-head in KVM virtualized environment. VIP stops working, but the node IP still works.
> 
> 1. Enable VLAN with NIC harvester-mgmt
> 2. Create VLAN 1
> 3. Reboot system
> 4. Wait for system to be ready showing VIP
> 5. Try to go to web UI at VIP
>    [supportbundle_d8ce0447-ea2a-40b7-880d-937953c890b5_2021-12-15T18-42-59Z.zip](https://github.com/harvester/harvester/files/7721956/supportbundle_d8ce0447-ea2a-40b7-880d-937953c890b5_2021-12-15T18-42-59Z.zip)

It's very likely the same problem as the issue https://github.com/harvester/harvester/issues/1681. Could you try to delete the pod of the kube-vip?

**Comment by TachunLin** [comment]
### Test Scenario 1:  PASS

1. Enable VLAN with NIC harvester-mgmt
2. Create VLAN 1
3. Disable VLAN 
4. Press ctrl+r to reload page 
5. Enable VLAN again
6. Without reboot
6. Check if you can access VIP and each node IP 

---

### Please take care with above steps
**Verify on previous harvester version (before 12/16) we haven't fixed it, if you follow above steps** 

> You are not be able to access harvester VIP nor each node IP 

> Even though we reboot all the harvester node, the issue still remains
 
* Lost VIP and node IP connectivity
```
davidtclin@localhost:~/Documents/Harvester/tools> ping 10.84.44.114
PING 10.84.44.114 (10.84.44.114) 56(84) bytes of data.
^C
--- 10.84.44.114 ping statistics ---
48 packets transmitted, 0 received, 100% packet loss, time 48116ms

davidtclin@localhost:~/Documents/Harvester/tools> ping 10.84.44.115
PING 10.84.44.115 (10.84.44.115) 56(84) bytes of data.
^C
--- 10.84.44.115 ping statistics ---
36 packets transmitted, 0 received, 100% packet loss, time 35817ms

davidtclin@localhost:~/Documents/Harvester/tools> ping 10.84.44.116
PING 10.84.44.116 (10.84.44.116) 56(84) bytes of data.
^C
--- 10.84.44.116 ping statistics ---
22 packets transmitted, 0 received, 100% packet loss, time 21496ms

davidtclin@localhost:~/Documents/Harvester/tools> ping 10.84.44.181
PING 10.84.44.181 (10.84.44.181) 56(84) bytes of data.
^C
--- 10.84.44.181 ping statistics ---
44 packets transmitted, 0 received, 100% packet loss, time 44027ms

```
---

**Verify on latest version `master-26deb4cd-head` we already have the fix, when you are above test steps** 

> Now you can access harvester VIP nor each node IP after disable/enable virtual network 

```
davidtclin@localhost:~/Documents/Harvester> ping 192.168.122.209
PING 192.168.122.209 (192.168.122.209) 56(84) bytes of data.
64 bytes from 192.168.122.209: icmp_seq=1 ttl=64 time=0.321 ms
^C
--- 192.168.122.209 ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 0.321/0.321/0.321/0.000 ms
davidtclin@localhost:~/Documents/Harvester> ping 192.168.122.33
PING 192.168.122.33 (192.168.122.33) 56(84) bytes of data.
64 bytes from 192.168.122.33: icmp_seq=1 ttl=64 time=0.347 ms
^C
--- 192.168.122.33 ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 0.347/0.347/0.347/0.000 ms
davidtclin@localhost:~/Documents/Harvester> ping 192.168.122.189
PING 192.168.122.189 (192.168.122.189) 56(84) bytes of data.
64 bytes from 192.168.122.189: icmp_seq=1 ttl=64 time=0.215 ms
^C
--- 192.168.122.189 ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 0.215/0.215/0.215/0.000 ms
davidtclin@localhost:~/Documents/Harvester> ping 192.168.122.52
PING 192.168.122.52 (192.168.122.52) 56(84) bytes of data.
64 bytes from 192.168.122.52: icmp_seq=1 ttl=64 time=0.262 ms
64 bytes from 192.168.122.52: icmp_seq=2 ttl=64 time=0.167 ms
64 bytes from 192.168.122.52: icmp_seq=3 ttl=64 time=0.176 ms
^C
```

**Comment by TachunLin** [comment]
### Test Scenario 2:  PASS (bare-metal machine)

1. Enable VLAN with NIC harvester-mgmt
2. Create VLAN 1
3. Disable VLAN 
4. Enable VLAN again
5. shutdown node 3, 2, 1 server machine 
6. Wait for 15 minutes
7. Power on node 1 server machine, wait for 20 seconds
8. Power on node 2 server machine, wait for 20 seconds
9. Power on node 3 server machine
10. Check if you can access VIP and each node IP 
---

Verified on version `master-5e8d9764-head`, this issue `did not happen` 

> We can access VIP and node IP after power on all nodes machine in sequence

### Test Information
* Test Environment: 3 nodes harvester on provo bare machines
* Harvester version: master-5e8d9764-head (12/16)

### Test Scenario 2:  PASS (local kvm machine)

Verified fixed on version `master-935a1670`, this issue `did not happen` 

### Test Information
* Test Environment: 3 nodes harvester on provo bare machines
* Harvester version: master-5e8d9764-head (12/16)
